src/include/miopen/fusion/solvers.hpp

```diff
@@ -0,0 +1,356 @@
+            return {bn_op.mode,
+                    bn_op.input_desc,
+                    out_desc,
+                    bn_op.base_desc,
+                    not_used, // expAvgFactor filler
+                    not_used,
+                    true /* resultSave*/,
+                    bn_op.runningMeanVar};
+        }
+        else if(dir == miopen::batchnorm::Direction::Backward)
+        {
+            const auto& bn_op =
+                dynamic_cast<BatchNormBwdTrainFusionOpDescriptor&>(*fusion_plan_desc->op_map[idx]);
+            miopen::TensorDescriptor out_desc;
+            bn_op.GetOutputDesc(out_desc);
+            return {bn_op.mode,
+                    bn_op.input_desc,
+                    out_desc,
+                    bn_op.input_desc,
+                    {} /*bn_op.base_desc*/,
+                    not_used,
+                    bn_op.useBatchStats /*useSaved*/};
+        }
+        else
+            MIOPEN_THROW(miopenStatusNotImplemented);
+    }
+};
+
+struct FusionContext : miopen::ExecutionContext
+{
+    FusionDescription problem;
+    FusionContext(FusionPlanDescriptor* ptr_desc, Handle& handle)
+        : ExecutionContext(&handle), problem(ptr_desc)
+    {
+    }
+    void GetNetworkConfig(std::string& net_config)
+    {
+        for(const auto& op : problem.fusion_plan_desc->op_map)
+        {
+            if(op->kind() == miopenFusionOpConvForward)
+            {
+                const auto prob = problem.GetConvProblem(op->GetIdx(), conv::Direction::Forward);
+                net_config += prob.conv_problem.BuildConfKey().ToString();
+            }
+            else if(op->kind() == miopenFusionOpBatchNormInference)
+            {
+                const auto prob = problem.GetBnProblem(
+                    op->GetIdx(), miopen::batchnorm::Direction::ForwardInference);
+                net_config += prob.MakeNetworkConfig().ToString();
+            }
+            else if(op->kind() == miopenFusionOpBatchNormFwdTrain)
+            {
+                const auto prob = problem.GetBnProblem(
+                    op->GetIdx(), miopen::batchnorm::Direction::ForwardTraining);
+                net_config += prob.MakeNetworkConfig().ToString();
+            }
+            else if(op->kind() == miopenFusionOpBatchNormBwdTrain)
+            {
+                const auto prob =
+                    problem.GetBnProblem(op->GetIdx(), miopen::batchnorm::Direction::Backward);
+                net_config += prob.MakeNetworkConfig().ToString();
+            }
+            else
+            {
+                op->GetNetworkConfig(net_config, this->GetStream());
+            }
+        }
+        MIOPEN_LOG_I2(net_config);
+    }
+
+    ConvolutionContext GetConvContext(size_t idx, conv::Direction dir) const
+    {
+        const auto conv_prob = problem.GetConvProblem(idx, dir);
+        if(dir == conv::Direction::Forward)
+        {
+            auto ctx                       = ConvolutionContext{conv_prob};
```

Comment:
Fixed.

Reply:
> I would also suggest changing it from legacy style. For old style API you can just make std::tuple<ExecutionContext, FusionDescription> and use it as a legacy context type.

@DrizztDoUrden  The issue with trying to do that is then I cant use `FindSolution` in the `find_solution.hpp` which requires `problem` to be a member of the Context. Since `ConvolutionContext` follows the same design as `FusionContext` I am not sure why I would want to deviate unless we would like to change both, which is best addressed in a separate PR.

Please elaborate if I have missed the point.